Fix fancy boxes to fold like regular boxes

coiax opened the pull request:
:cl: coiax
fix: Folding a "fancy box" like donut or candles will now put cardboard in your hand, rather than on the floor.
fix: You can always see how many pickles are in a jar of pickles, or rolling papers there are in a pack of rolling papers, since it's not possible to close them.
/:cl:

Currently, folding a "fancy box", one where the contents are visible in the icon, like a donut or candle box, the cardboard will drop on the floor.

The `attack_self()` proc between the two types of boxes have been unified.

In addition, the `is_open` for fancy boxes has been expanded to include the "always open" option, where the contents of the box are always countable on examine, and the box does not have a closed appearance. Candle boxes previously special cased the `attack_self()` proc, and with this addition, no longer need to.

## Why It's Good For The Game

Consistency in box interaction. Cardboard in hand. Always can count pickles.

coiax committed:
Fix fancy boxes to fold like regular boxes

:cl: coiax
fix: Folding a "fancy box" like donut or candles will now put cardboard
in your hand, rather than on the floor.
fix: You can always see how many pickles are in a jar of pickles, or
rolling papers there are in a pack of rolling papers, since it's not
possible to close them.
/:cl:

Currently, folding a "fancy box", one where the contents are visible in
the icon, like a donut or candle box, the cardboard will drop on the
floor.

The `attack_self()` proc between the two types of boxes have been
unified.

In addition, the `is_open` for fancy boxes has been expanded to
include the "always open" option, where the contents of the box are
always countable on examine, and the box does not have a closed
appearance. Candle boxes previously special cased the `attack_self()`
proc, and with this addition, no longer need to.

san7890 commented on code/game/objects/items/storage/boxes/_boxes.dm:
```diff
@@ -47,8 +47,8 @@
 		return
 
+	var/obj/item/I = new foldable(user.drop_location())
 	balloon_alert(user, "folded")
-	var/obj/item/I = new foldable
 	qdel(src)
 	user.put_in_hands(I)
```
clean up this single-letter var into something longer/more appropriate

san7890 commented on code/game/objects/items/storage/fancy.dm:
```diff
@@ -56,26 +60,33 @@
 		return
-	new fold_result(user.drop_location())
+	var/obj/item/I = new foldable(user.drop_location())
 	balloon_alert(user, "folded")
-	user.put_in_active_hand(fold_result)
+	// Gotta delete first, so then the cardboard appears in the same hand
 	qdel(src)
+	user.put_in_hands(I)
```
ditto clean up single letter var

san7890 commented on code/game/objects/items/storage/fancy.dm:
```diff
@@ -23,10 +27,10 @@
-	var/obj/fold_result = /obj/item/stack/sheet/cardboard
+	/// Whether the container is open, always open, or closed
+	var/is_open = CONTAINER_CLOSED
+	/// What material do we get when we fold this box?
+	var/foldable = /obj/item/stack/sheet/cardboard
```
should still be named `foldable_result` or something, since `foldable` alone implies it's a boolean to someone who isn't being careful

san7890 commented on code/game/objects/items/storage/fancy.dm:
```diff
@@ -23,10 +27,10 @@
-	var/is_open = FALSE
-	/// What this container folds up into when it's empty.
-	var/obj/fold_result = /obj/item/stack/sheet/cardboard
+	/// Whether the container is open, always open, or closed
+	var/is_open = CONTAINER_CLOSED
```
this could also be something like `open_status` since it's more than just a standard boolean

san7890 reviewed (changes requested):
few nits

coiax committed:
Merge branch 'master' into unifying-fancy-and-regular-box-interaction

coiax committed:
Applied suggested changes for variable names

- `foldable` for boxes and fancy boxes is now renamed to
  `foldable_result`, the outcome of what you get if you attempt to fold
   the box.
- Some single letter variables have been cleaned up.
- `is_open` has been renamed to `open_status` since it is no longer a
  boolean.
